GRSIProof/ExampleEventSelector.C

```diff
@@ -4,24 +4,25 @@
 
 #include "TH1.h"
 
-//need to define these histograms either globally or as members of the class
-TH1D* gE;
-TH1D* gE_b;
+void ExampleEventSelector::CreateHistograms() {
+   fH1["gE"] = new TH1D("gE","#gamma Singles",12000,0,3000);
+   fH1["gE_b"] = new TH1D("gE_b","#gamma Singles in rough #beta coincidence",12000,0,3000);
 
-void ExampleEventSelector::CreateHistograms()
-{
-   gE = new TH1D("gE","#gamma Singles",12000,0,3000);
-   gE_b = new TH1D("gE_b","#gamma Singles in rough #beta coincidence",12000,0,3000);
-   GetOutputList()->AddAll(gDirectory->GetList());
+	for(auto it : fH1) {
+		GetOutputList()->Add(it.second);
+	}
+	for(auto it : fH2) {
+		GetOutputList()->Add(it.second);
+	}
+	for(auto it : fHSparse) {
+		GetOutputList()->Add(it.second);
+	}
 }
 
-void ExampleEventSelector::FillHistograms()
-{
+void ExampleEventSelector::FillHistograms() {
    for(auto i = 0; i < fGrif->GetMultiplicity(); ++i){
-      gE->Fill(fGrif->GetGriffinHit(i)->GetEnergy());
+      fH1["gE"]->Fill(fGrif->GetGriffinHit(i)->GetEnergy());
```

Comment:
I'm assuming the string comparison + map lookup doesn't slow this down too much? I'm not sure myself how to make it this robust without doing what you did, just curious.